From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Igor Pecovnik <igorpecovnik@users.noreply.github.com>
Date: Wed, 2 Nov 2022 06:58:31 +0100
Subject: [ARCHEOLOGY] Add Radxa ROCK Pi 4C Plus (#4129)

> X-Git-Archeology: > recovered message: > * Add Rockpi 4C+
> X-Git-Archeology: > recovered message: > * add newer patches:
> X-Git-Archeology: > recovered message: > - apply patches from RadxaNaoki <noaki@radxa.com>
> X-Git-Archeology: > recovered message: > - ref 5652d6f9c2a4e2c50ac1040c0388859381b0616f: arm64: dts: rockchip: add gmac for ROCK 4C+ (#17)
> X-Git-Archeology: > recovered message: > - apply latest patches from Marcone <marco.nelissen@gmail.com>
> X-Git-Archeology: > recovered message: > - ref e39da181b099a86f5440ed1c08fc699d4a65aed7: arm64: dts: rockchip: Make Rock Pi 4C+ ethernet actually work
> X-Git-Archeology: > recovered message: > - ref 8c519612eeb852268371b3c596eec4622460e5bf: arm64: dts: rockchip: add Rock Pi 4C+ LEDs
> X-Git-Archeology: > recovered message: > - ref 726441afe687ea6059bac37f145ff204b75430f7: arm64: dts: rockchip: enable Rock Pi 4C+ wifi
> X-Git-Archeology: > recovered message: > - ref caad7bef3b5558e2598480c0a0089e2ac3cedcdd: arm64: dts: rockchip: enable Rock Pi 4C+ nvme
> X-Git-Archeology: > recovered message: > - ref 6cdc3215d1691bcd2a407a30e88efa0cec179eb2: arm64: dts: rockchip: enable Rock Pi 4C+ OTG
> X-Git-Archeology: > recovered message: > * default top USB3 port to host mode
> X-Git-Archeology: > recovered message: > * get 2nd HDMI port to work at least. :)
> X-Git-Archeology: > recovered message: > * revert OTG USB3.1 port back to otg mode
> X-Git-Archeology: > recovered message: > * [rockpi4c+] update board def.  add usb host overlay.
> X-Git-Archeology: > recovered message: > * remove redundant HDMI clocks and voltages
> X-Git-Archeology: > recovered message: > * enable test builds for Rock PI 4 C+
> X-Git-Archeology: > recovered message: > * [rockpi4c+] add USB host overlay
> X-Git-Archeology: > recovered message: > Co-authored-by: Joe Khoobyar <fourheads@gmail.com>
> X-Git-Archeology: - Revision 9780da320eeb45832ddb680959646fcc5a6a38c3: https://github.com/armbian/build/commit/9780da320eeb45832ddb680959646fcc5a6a38c3
> X-Git-Archeology:   Date: Wed, 02 Nov 2022 06:58:31 +0100
> X-Git-Archeology:   From: Igor Pecovnik <igorpecovnik@users.noreply.github.com>
> X-Git-Archeology:   Subject: Add Radxa ROCK Pi 4C Plus (#4129)
> X-Git-Archeology:
> X-Git-Archeology: - Revision af24261cad8aac0fe13efd57449a82ab1b33e868: https://github.com/armbian/build/commit/af24261cad8aac0fe13efd57449a82ab1b33e868
> X-Git-Archeology:   Date: Wed, 02 Nov 2022 10:45:35 -0400
> X-Git-Archeology:   From: Joe Khoobyar <joekhoobyar@users.noreply.github.com>
> X-Git-Archeology:   Subject: AR-1301: add edge and desktop (#4377)
> X-Git-Archeology:
> X-Git-Archeology: - Revision 92f1a22d76b987afa7ba555d5b509adc51d689e7: https://github.com/armbian/build/commit/92f1a22d76b987afa7ba555d5b509adc51d689e7
> X-Git-Archeology:   Date: Fri, 16 Dec 2022 13:38:13 +0100
> X-Git-Archeology:   From: Igor Pecovnik <igorpecovnik@users.noreply.github.com>
> X-Git-Archeology:   Subject: Re-add rockchip64 6.0 patches (#4575)
> X-Git-Archeology:
> X-Git-Archeology: - Revision 34ae84fac5d0b66a1ab2d1e51534b7beb13ef245: https://github.com/armbian/build/commit/34ae84fac5d0b66a1ab2d1e51534b7beb13ef245
> X-Git-Archeology:   Date: Fri, 05 May 2023 14:22:00 +0200
> X-Git-Archeology:   From: amazingfate <liujianfeng1994@gmail.com>
> X-Git-Archeology:   Subject: bump rockchip64 edge to v6.3
> X-Git-Archeology:
---
 arch/arm64/boot/dts/rockchip/overlay/Makefile                           |  1 +
 arch/arm64/boot/dts/rockchip/overlay/README.rockchip-overlays           |  5 +++
 arch/arm64/boot/dts/rockchip/overlay/rockchip-rockpi4cplus-usb-host.dts | 16 ++++++++++
 3 files changed, 22 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/overlay/Makefile b/arch/arm64/boot/dts/rockchip/overlay/Makefile
index 3c2264543bef..41c323473d50 100644
--- a/arch/arm64/boot/dts/rockchip/overlay/Makefile
+++ b/arch/arm64/boot/dts/rockchip/overlay/Makefile
@@ -6,6 +6,7 @@ dtbo-$(CONFIG_ARCH_ROCKCHIP) += \
 	rockchip-rk3328-opp-1.4ghz.dtbo \
 	rockchip-rk3328-opp-1.5ghz.dtbo \
 	rockchip-rk3399-opp-2ghz.dtbo \
+	rockchip-rockpi4cplus-usb-host.dtbo \
 	rockchip-spi-jedec-nor.dtbo \
 	rockchip-spi-spidev.dtbo \
 	rockchip-uart4.dtbo \
diff --git a/arch/arm64/boot/dts/rockchip/overlay/README.rockchip-overlays b/arch/arm64/boot/dts/rockchip/overlay/README.rockchip-overlays
index 0be01217af7a..17b21826c79a 100644
--- a/arch/arm64/boot/dts/rockchip/overlay/README.rockchip-overlays
+++ b/arch/arm64/boot/dts/rockchip/overlay/README.rockchip-overlays
@@ -48,6 +48,11 @@ WARNING! Not officially supported by Rockchip!!!
 Adds the 2GHz big and 1.5 GHz LITTLE opps for overclocking
 WARNING! Not officially supported by Rockchip!!!
 
+### rockpi4cplus-usb-host
+
+Switches the top USB 3.0 port to host mode.
+WARNING! Not officially supported by Rockchip!!!
+
 ### spi-jedec-nor
 
 Activates MTD support for JEDEC compatible SPI NOR flash chips on SPI bus
diff --git a/arch/arm64/boot/dts/rockchip/overlay/rockchip-rockpi4cplus-usb-host.dts b/arch/arm64/boot/dts/rockchip/overlay/rockchip-rockpi4cplus-usb-host.dts
new file mode 100644
index 000000000000..654ae888f5c4
--- /dev/null
+++ b/arch/arm64/boot/dts/rockchip/overlay/rockchip-rockpi4cplus-usb-host.dts
@@ -0,0 +1,16 @@
+//For RockPI 4C+:  Change the top USB3.0 port to host mode
+
+/dts-v1/;
+/plugin/;
+
+/ {
+	compatible = "radxa,rockpi4c-plus", "radxa,rockpi4", "rockchip,rk3399";
+
+	fragment@0 {
+		target = <&usbdrd_dwc3_0>;
+		__overlay__ {
+			dr_mode = "host";
+		};
+	};
+};
+
-- 
Armbian

